### ISSUE TITLE
MetadataFile.java:45-46: Add a unit test for this class,...

### ISSUE THREAD

**Issue by 0pdd** [artipie/rpm-adapter#78]
The puzzle `69-b66a2919` from #69 has to be resolved: 

https://github.com/artipie/rpm-adapter/blob/493153a41dfc9dcae2bb43d8af290adf7d7acf47/src/main/java/com/artipie/rpm/pkg/MetadataFile.java#L45-L46

The puzzle was created by @g4s8 on 08-Apr-20. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) please, pay attention to this issue

**Comment by 0crat** [comment]
The job [#78](https://github.com/artipie/rpm-adapter/issues/78) assigned to @ammaratef45[/z](https://www.0crat.com/u/ammaratef45), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/eda9f559-66a3-425f-a212-27aee3aaeed5); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by ammaratef45** [comment]
@g4s8 in order to try the `accept` method I need to path a `Mata` object, the only implementation I found is the private class `Header` from `FilePackage.java`
What's the correct way to inject some test `Meta` objects?

**Comment by g4s8** [comment]
@ammaratef45 I'd create some custom Hamcrest matcher to verify `Meta` headers, e.g.:
```java
class MetaHasHeader extends TypeSafeMatcher<Meta>, PackageOutput {
  private final Header.HeaderTag tag;
  private final Matcher<String> matcher;

  public boolean matchesSafely(Meta meta) {
    return this.matcher.matches(meta.header(tag));
  }
 
  public void accept(Meta meta) {
    MatcherAssert.assertThat(meta, this);
  }
}
```
and use it in tests like:
```java
@Test
void someTest() {
  new FilePackage(rpm).accept(new MetaHasHeader(Header.HeaderTag, "value"));
}
```

maybe you can find some better solution, like split output and matcher or create output matcher.

**Comment by ammaratef45** [comment]
@g4s8 I've tried to figure it out but, apparently, I couldn't make it good enough. I'll learn from the solution then.

**Comment by ammaratef45** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/artipie/rpm-adapter/issues/78#issuecomment-612201460))

@ammaratef45 The user @ammaratef45[/z](https://www.0crat.com/u/ammaratef45) resigned from [#78](https://github.com/artipie/rpm-adapter/issues/78), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
The job [#78](https://github.com/artipie/rpm-adapter/issues/78) assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/b4f1742f-1808-4589-a5e9-6abacb2c0249); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by victornoel** [comment]
@0crat wait for #100 to be merged

**Comment by 0pdd** [comment]
The puzzle `69-b66a2919` has disappeared from the source code, that's why I closed this issue.

**Comment by 0pdd** [comment]
@0pdd the puzzle [#106](https://github.com/artipie/rpm-adapter/issues/106) is still not solved.

**Comment by victornoel** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/artipie/rpm-adapter/issues/78#issuecomment-612967598))

@victornoel This is what I know about this job in CT2E6TK9B, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#78](https://github.com/artipie/rpm-adapter/issues/78) is [in scope](http://www.zerocracy.com/policy.html#14) for 5days
  * The role is `DEV`
  * The job is assigned to @victornoel[/z](https://www.0crat.com/u/victornoel) for [2days](http://www.zerocracy.com/policy.html#8)
  * There is a monetary reward attached
  * @victornoel[/z](https://www.0crat.com/u/victornoel) will get 50% of their hourly rate for this job
  * The job has an [impediment](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 30 minutes/points
  * These users are banned and won't be assigned:
    * @0pdd[/z](https://www.0crat.com/u/0pdd): This user reported the ticket
    * @ammaratef45[/z](https://www.0crat.com/u/ammaratef45): User was resigned from the ticket
  * Job [footprint](https://www.0crat.com/footprint/CT2E6TK9B?q=%7Bjob%3A%27gh%3Aartipie%2Frpm-adapter%2378%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by victornoel** [comment]
@g4s8 0crat didn't noticed the issue was closed, what should be done to trigger it?

**Comment by victornoel** [comment]
@g4s8 please can you look at that and notify 0crat to go on with the process?

**Comment by g4s8** [comment]
@0crat out

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @victornoel[/z](https://www.0crat.com/u/victornoel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0pdd** [comment]
@0pdd the puzzle [#206](https://github.com/artipie/rpm-adapter/issues/206) is still not solved; solved: [#106](https://github.com/artipie/rpm-adapter/issues/106).

**Comment by 0pdd** [comment]
@0pdd all 2 puzzles are solved here: [#106](https://github.com/artipie/rpm-adapter/issues/106), [#206](https://github.com/artipie/rpm-adapter/issues/206).
